cmd/gapic-showcase/qualifier/scenario.go

```diff
@@ -0,0 +1,359 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qualifier
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	packr "github.com/gobuffalo/packr/v2"
+	trace "github.com/google/go-trace"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Constants pertaining to the syntax of include files. Include files are akin to symbolic links:
+// they reference content that will be copied at run-time from the indicated location to the include
+// file's location under the indicated names. For example, an include file called "include.boren"
+// with contents "api-common-protos/google/" will be replaced at run-time with a directory called
+// "boren" whose contents will be a copy of everything under "api-common-protos/google/"
+//
+// We need some mechanism for including common protos because the protos under test may reference
+// them. This particular scheme is also useful in that it allows us to explicitly pass to the
+// generator only the protos explicitly listed in the acceptance check, rather than also having to
+// include all the protos that are transitively included.
+const (
+	// In a qualifier suite directory, any file that begins with this prefix is considered an
+	// include file. The rest of the filename after the prefix is the destination filename to
+	// which the files referenced by the content of the include file will be copied.
+	includeFilePrefix = "include."
+
+	// In the contents of an include file, this symbol will be treated as the directory
+	// separator. At run time, it will be replaced the appropriate os.PathSeparator.
+	includeFileDirectorySeparator = '/'
+)
+
+type Scenario struct {
+	// Every element in `files` corresponds to a file in `fileBox` under
+	// a directory called `name`, which is NOT part of the string
+	// element itself.
+	name        string              // name of the scenario
+	timestamp   string              // timestamp for debug and trace messages
+	files       []string            // a flat list of files in the scenario
+	fileBox     *packr.Box          // the files explicitly specified in the scenario
+	schemaBox   *packr.Box          // additional files that may be referenced by include files in a scenario
+	sandbox     string              // the sandbox created to run this scenario
+	filesByType map[string][]string // scenario files partitioned by file type
+
+	generator    *Generator
+	showcasePort int // the port of the running showcase server against which samples will run
```

Comment:
I'm sorry, I didn't ask the question I really wanted the answer to!

Can we rename it to just `port`? The `showcase` part is kind of a given because this code is in showcase and also makes the variable rigid, meaning we will have to refactor the code if we ever wanted to point it at something else other than showcase.

Reply:
Oh, sorry! I didn't pick up on the implied question. OK, I'll look at renaming it. Maybe `servicePort`?

Reply:
Sure that's fine. Thanks